prov/util/src/util_wait.c

```diff
@@ -424,3 +428,214 @@ int ofi_wait_fd_open(struct fid_fabric *fabric_fid, struct fi_wait_attr *attr,
 	free(wait);
 	return ret;
 }
+
+static void util_wait_yield_signal(struct util_wait *util_wait)
+{
+	(container_of(util_wait, struct util_wait_yield,
+			util_wait))->signal = 1;
+}
+
+static int util_wait_yield_try(struct util_wait *wait)
+{
+	struct ofi_wait_fid_entry *fid_entry;
+	struct util_wait_yield *wait_yield;
+	int ret = 0;
+
+	wait_yield = container_of(wait, struct util_wait_yield, util_wait);
+	fastlock_acquire(&wait_yield->lock);
+	wait_yield->signal = 0;
+	dlist_foreach_container(&wait_yield->fid_list, struct ofi_wait_fid_entry,
+				fid_entry, entry) {
+		ret = fid_entry->wait_try(fid_entry->fid);
+		if (ret != FI_SUCCESS)
+			goto out;
+	}
+
+	if (!ret && !wait_yield->signal)
+		ret = -FI_EAGAIN;
+
+out:
+	fastlock_release(&wait_yield->lock);
+	return ret;
+}
+
+static int util_wait_yield_run(struct fid_wait *wait_fid, int timeout)
+{
+	struct util_wait_yield *wait;
+	int ret;
+
+	wait = container_of(wait_fid, struct util_wait_yield, util_wait.wait_fid);
+	while (1) {
+		ret = wait->util_wait.wait_try(&wait->util_wait);
+		if (ret != -FI_EAGAIN)
+			return ret;
```

Comment:
But where's the yield call?

Reply:
Sorry just saw this one. I don't think I understand?

Reply:
We're busy spinning in a while (1) loop here, but we don't ever actually yield the processor. You know, like calling pthread_yield() / sched_yield().

Reply:
Where would be the best place for this call?